x/fungible/keeper/msg_server_update_system_contract.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	"context"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/protocol-contracts/pkg/contracts/zevm/systemcontract.sol"
+	"github.com/zeta-chain/protocol-contracts/pkg/contracts/zevm/zrc20.sol"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+	zetaObserverTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k Keeper) UpdateSystemContract(goCtx context.Context, msg *types.MsgUpdateSystemContract) (*types.MsgUpdateSystemContractResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if msg.Creator != k.zetaobserverKeeper.GetParams(ctx).GetAdminPolicyAccount(zetaObserverTypes.Policy_Type_deploy_fungible_coin) {
```

Comment:
Shouldn't we rename the policty type `Policy_Type_deploy_fungible_coin`, it seems to be of a broader scope to me like the admin of system contracts.

Reply:
We can use the same group or policy for different Policy Types.

Reply:
so in this particular case we create a new policy type? @kingpinXD

Reply:
Yes, I think creating a new policy type might be a good idea, Or we can just rename the existing policy so that the identifier includes both the functions it's responsible for.

Reply:
actually let's keep it as it is. We can revise policy types in a separate PR.

Reply:
I thing using the same policy type for deploy and update is fine , might want to rename the identifier as suggested in the original comment , that should be enough

Reply:
https://github.com/zeta-chain/node/issues/932